### PR TITLE
[rag eval] add concept of feedback stage

### PR DESCRIPTION
**Pull request by chejennifer**
- Add a new FeedbackStage enum to be used when deciding what to display in the query section and feedback sections
- Changes to make RIG eval tool work with this new concept
- only change made for RAG was displaying the rag calls in the query section when feedback stage is the CALLS stage. Actual navigation for RAG will come later

### CODE REVIEW

File: static/js/apps/eval_retrieval_generation/util.ts [hunk 1]
```diff
@@ -66,3 +68,12 @@ export function processTableText(text: string): string {
   // created
   return text.replace(TABLE_DIVIDER_PATTERN, tableDivider);
 }
+
+// Get the first feedback stage to show for an eval type
+export function getFirstFeedbackStage(evalType: EvalType): FeedbackStage {
```
Comment (shifucun) [inline static/js/apps/eval_retrieval_generation/util.ts:73]: curious will there be function like getSecondFeedbackStage later?

**Review by shifucun — APPROVED** [review]
very clean and structured!

File: static/js/apps/eval_retrieval_generation/util.ts (hunk 1 above)
Comment (chejennifer) [reply static/js/apps/eval_retrieval_generation/util.ts:73]: Some ideas I'm playing with:
- getNextFeedbackStage(evalType, feedbackStage) & getPrevFeedbackStage(evalType, feedbackStage)
- turn feedbackNavigation into RigFeedbackNavigation and RagFeedbackNavigation and they each have the control over setting the feedback stage
- have a config that is a list of stages and as next or prev is clicked, we move through the list of stages

**Comment by chejennifer** [comment]
thanks for the review!

File: static/js/apps/eval_retrieval_generation/util.ts (hunk 1 above)
Comment (shifucun) [reply static/js/apps/eval_retrieval_generation/util.ts:73]: I see. So sounds like this needs more work.

**Pull request merged**